runtime/vm/stringhelpers.cpp

```diff
@@ -396,41 +332,39 @@ copyFromStringIntoUTF8(J9VMThread * vmThread, j9object_t string, char * dest)
  * @param[in] stringFlags the flag to determine performing '.' --> '/'
  * @param[in] prependStr the string to be prepended before the string object to be copied
  * 				it can't be NULL but can be an empty string ""
+ * @param[in] prependStrLength The length of prependStr as computed by strlen.
  * @param[in] buffer the buffer for the string
  * @param[in] bufferLength the buffer length, not expected to larger than 64K
+ * @param[out] utf8Length If not NULL returns the computed length (in bytes) of the copied UTF8 string in the buffer excluding the NULL terminator.
  *
  * @return a char pointer to the string (with NULL termination)
  */
 char*
-copyStringToUTF8WithMemAlloc(J9VMThread *vmThread, j9object_t string, UDATA stringFlags, const char *prependStr, char *buffer, UDATA bufferLength)
+copyStringToUTF8WithMemAlloc(J9VMThread *vmThread, j9object_t string, UDATA stringFlags, const char *prependStr, UDATA prependStrLength, char *buffer, UDATA bufferLength, UDATA *utf8Length)
 {
 	Assert_VM_notNull(prependStr);
 	Assert_VM_notNull(string);
 
 	char *strUTF = NULL;
-	UDATA stringLen = getStringUTF8Length(vmThread, string);
-	UDATA prependStrLen = strlen(prependStr);
-	UDATA length = stringLen + prependStrLen + 1;
+	UDATA stringLen = J9VMJAVALANGSTRING_LENGTH(vmThread, string) * 3;
+	UDATA length = stringLen + prependStrLength + 1;
 
 	PORT_ACCESS_FROM_VMC(vmThread);
 
 	if (length > bufferLength) {
 		strUTF = (char *)j9mem_allocate_memory(length, OMRMEM_CATEGORY_VM);
 	} else {
 		strUTF = buffer;
-		length = bufferLength;
 	}
 	if (NULL != strUTF) {
-		if (0 < prependStrLen) {
-			memcpy(strUTF, prependStr, prependStrLen);
+		if (0 < prependStrLength) {
+			memcpy(strUTF, prependStr, prependStrLength);
 		}
-		if (UDATA_MAX == copyStringToUTF8Helper(
-			vmThread, string, TRUE, stringFlags, (U_8 *)(strUTF + prependStrLen), length - prependStrLen)
-		) {
-			if (buffer != strUTF) {
-				j9mem_free_memory(strUTF);
-			}
-			return NULL;
+
+		UDATA computedUtf8Length = copyStringToUTF8Helper(vmThread, string, stringFlags, (U_8 *)(strUTF + prependStrLength));
```

Comment:
Defining a variable not at the beginning of the code block will cause compilation failure in certain platforms (zOS).

Reply:
> Defining a variable not at the beginning of the code block will cause compilation failure in certain platforms (zOS).

This is a statement valid for source files compiled with a C compiler. This is a C++ file. There are other examples of declarations in the middle of a scoped block in the same file here:

https://github.com/eclipse/openj9/blob/b8d557446f0eb50dca4ade4b0ff5002f30193d52/runtime/vm/stringhelpers.cpp#L176-L180

Reply:
Agreed though the example is not accurate in which the variables were defined at beginning of a code block `{}`.
For the sake of consistency and avoiding confusion, I suggest moving the variable definition just four lines above.

Reply:
Fixed in bced3dc